src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/CraftingRequestResolver.java

```diff
@@ -0,0 +1,164 @@
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static com.minecolonies.api.util.RSConstants.CONST_CRAFTING_RESOLVER_PRIORITY;
+
+/**
+ * ----------------------- Not Documented Object ---------------------
+ */
+public class CraftingRequestResolver extends AbstractRequestResolver<IDeliverable>
+{
+    public CraftingRequestResolver(
+                                     @NotNull final ILocation location,
+                                     @NotNull final IToken<?> token)
+    {
+        super(location, token);
+    }
+
+    @Override
+    public TypeToken<? extends IDeliverable> getRequestType()
+    {
+        return TypeToken.of(IDeliverable.class);
+    }
+
+    @Override
+    public boolean canResolve(@NotNull final IRequestManager manager, final IRequest<? extends IDeliverable> requestToCheck)
+    {
+        if (!manager.getColony().getWorld().isRemote)
+        {
+            Colony colony = (Colony) manager.getColony();
+            final ILocation requesterLocation = requestToCheck.getRequester().getRequesterLocation();
+            final AbstractBuilding building = colony.getBuilding(requesterLocation.getInDimensionLocation());
+            if(building instanceof AbstractBuildingWorker)
+            {
+                final ItemStack stack = requestToCheck.getRequest().getResult();
+
+                if (ItemStackUtils.isEmpty(stack))
+                {
+                    return false;
+                }
+
+                if ((((AbstractBuildingWorker) building).canCraft(stack)))
+                {
+                    return true;
+                }
+            }
+        }
+
+        return false;
+    }
+
+    @Nullable
+    @Override
+    @SuppressWarnings("squid:LeftCurlyBraceStartLineCheck")
+    /**
+     * Moving the curly braces really makes the code hard to read.
+     */
+    public List<IToken<?>> attemptResolve(
+                                        @NotNull final IRequestManager manager, @NotNull final IRequest<? extends IDeliverable> request)
+    {
+        if (manager.getColony().getWorld().isRemote)
+        {
+            return null;
+        }
+
+        final Colony colony = (Colony) manager.getColony();
+        final ILocation requesterLocation = request.getRequester().getRequesterLocation();
+        final AbstractBuilding building = colony.getBuilding(requesterLocation.getInDimensionLocation());
+        if(canResolve(manager, request) && building instanceof AbstractBuildingWorker)
+        {
+            final ItemStack stack = request.getRequest().getResult();
+            if(ItemStackUtils.isEmpty(stack))
+            {
+                return Lists.newArrayList();
+            }
+
+            final RecipeStorage storage = ((AbstractBuildingWorker) building).getFirstFullFillableRecipe(stack);
+
+            if(storage == null)
+            {
+                return null;
+            }
+            ((AbstractBuildingWorker) building).fullFillRecipe(storage);
```

Comment:
Please do not do this!
This method only is for detecting requirements.

Use this method to figure out which stacks you need or not, and create requests for them using the given Manager.
This method is not for performing the craft.

Reply:
okay, Understood makes sense